src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -313,8 +384,12 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 		 * in group and order by with aggregate operations.
 		 * see nodeAgg.c:build_pertrans_for_aggref(). In that case we use
 		 * sorted agg strategy, otherwise we use hash strategy.
+		 *
+		 * If the master query contains hll aggregate functions and the client set
+		 * hll.enable_hashagg to off, then we choose to use group aggregation.
 		 */
-		if (!enable_hashagg || !groupingIsHashable || hasDistinctAggregate)
+		if (!enable_hashagg || !groupingIsHashable || hasDistinctAggregate ||
+			UseGroupAggregateWithHLL(masterQuery))
```

Comment:
I would rather see this next to  `groupingIsHashable` and `hasDistinctAggregate` calculations.

Reply:
Added in that way to not decrease the performance of aggregates without hll type